torch/csrc/jit/serialization/export.cpp

```diff
@@ -1370,10 +1370,26 @@ void check_onnx_proto(const std::string& proto_string, bool full_check) {
     throw std::runtime_error("Invalid ONNX proto string.");
     return;
   }
+  // baseline check
   onnx::checker::check_model(model);
-
+  onnx::shape_inference::InferShapes(model);
+  // full check including strict shape inference check
+  // Note: onnx::checker::check_model doesn't
+  // include non-strict model onnx::shape_inference::InferShapes(model).
+  // That's why we need to write another onnx::checker::check_model(model,
+  // full_check) here.
   if (full_check) {
-    onnx::shape_inference::InferShapes(model);
+    try {
+      onnx::checker::check_model(model, full_check);
```

Comment:
I think we can directly run shape inference with strict mode:
```suggestion
      auto* schema_registry = onnx::OpSchemaRegistry::Instance();
      onnx::ShapeInferenceOptions options{
          /*check_type=*/True,
          /*error_mode=*/True};
      onnx::shape_inference::InferShapes(
          *model_proto, schema_registry, options);
```
Otherwise it runs onnx.checker twice.

Reply:
Nice catch